aau-giraf/weekplanner-OLDUI
# Feature/856v2

# Description
This PR makes progress toward solving the issue https://github.com/aau-giraf/weekplanner/issues/856


The work done on this branch extends the previous progress made on the branches:

Weekplanner branch: https://github.com/aau-giraf/weekplanner/tree/feature/856
Api_client branch: https://github.com/aau-giraf/api_client/tree/feature/856

### The branches had completed the following subtasks:

- [x] Add "Tilføj billede" button on the "Ny borger" page
- [x] Add "Tag billede" button on the "Ny borger" page
- [x] Add preview of the selected picture on the "Ny borger" page
- [x] When picture is added, still able to change to another picture


This PR is linked to two other PRs:
Web-api: [PR](https://github.com/aau-giraf/web-api/pull/302)
Api-client: [PR](https://github.com/aau-giraf/api_client/pull/124)

### The progress made on this branch are the following subtasks:
- [x] Retrieve picture from new_citizen_bloc and send to server
- [x] upload picture to database
- [x] retrieve user pictures from server
- [x] display user pictures on the Choose Citizen screen


This feature adds the ability for a user to contain a profile picture. The profile picture is optional and will be null by default.

When a citizen is created in the Weekplan app, there will be an optional field to upload a picture. 
![image](https://user-images.githubusercontent.com/73853586/202673454-7c6733ed-0a8f-4b25-b1b3-37b2749890a1.png)
The endpoints that have changed with this feature are:
/v2/Account/register
/v1/User
/v1/User/21/citizens


The citizen's pictures will also be displayed on the user-select screen of the guardian if a picture is available.
![image](https://user-images.githubusercontent.com/73853586/202674558-92981ff2-0980-4311-bef2-e881d0c75998.png)



## Type of change



# How Has This Been Tested?

Tested manually. Tests are yet to be implemented.

## How to test for yourself
This should be tested together with the web-api (https://github.com/aau-giraf/web-api/tree/feature/856v2) in this way:
- Get these changes to the web-api locally, then set up the database locally with these changes.
- change line 16 in pubspec.yaml in the weekplanner from `develop ` to `feature/856v2`
- run the database
- start the weekplanner
- Create a new citizen and upload a profile picture
- See the picture on the choose citizen screen



**Development Configuration**
* Flutter version: 2.0.5
* Dart version: 2.12.3

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have added tests that prove my fix is effective or that my feature works, if necessary
- [ ] I have Acceptance Tested this on an iOS device
- [x] I have Acceptance Tested this on an Android device
